```
---[p.160]-------------------------------

Running association scans for SNP(複数SNPと形質の関連解析)


> xtmp<- matrix(c(24,21,2,68,32,10),ncol=2)
> xtmp
     [,1] [,2]
[1,]   24   68
[2,]   21   32
[3,]    2   10
> chisq.test(xtmp)

	Pearson's Chi-squared test

data:  xtmp
X-squared = 4.0282, df = 2, p-value = 0.1334

Warning message:
In chisq.test(xtmp) :  カイ自乗近似は不正確かもしれません


宿題：SNP10001,10002, 10005,10008,10019,10020についてケースコントロール解析(カイ2乗検定)を行う。



> library(SNPassoc)
 要求されたパッケージ haplo.stats をロード中です 
 要求されたパッケージ survival をロード中です 
 要求されたパッケージ mvtnorm をロード中です 
 要求されたパッケージ parallel をロード中です 
> data(SNPs)
> mySNP<-setupSNP(SNPs,6:40,sep="")
> mySNP
     id casco    sex blood.pre    protein snp10001 snp10002 snp10003 snp10004 snp10005 snp10006 snp10007 snp10008 snp10009
1     1     1 Female      13.7  75640.523      T/T      C/C      G/G      G/G      G/G      A/A      C/C      C/C      A/A
2     2     1 Female      12.7  28688.215      T/T      A/C      G/G      G/G      A/G      A/A      C/C      C/C      A/G
3     3     1 Female      12.9  17279.591      T/T      C/C      G/G      G/G      G/G      A/A      C/C      C/C      A/A
4     4     1   Male      14.6  27253.988      C/T      C/C      G/G      G/G      G/G      A/A      C/C      C/C      A/A
5     5     1 Female      13.4  38066.569      T/T      A/C      G/G      G/G      G/G      A/A      C/C      C/C      A/G
6     6     1 Female      11.3   9872.460      T/T      C/C      G/G      G/G      G/G      A/A      C/C      C/C      A/A
7     7     1 Female      11.9  11132.903      T/T      A/C      G/G      G/G      A/G      A/A      C/C      C/C      A/G
8     8     1   Male




> myres1 <- WGassociation(protein~1,data=mySNP,model="all")
> 
> myres1
          comments    codominant dominant recessive overdominant log-additive
snp10001  -           0.00586    0.00516  0.01605   0.12306      0.00140     
snp10002  -           0.78555    0.93303  0.48704   0.87288      0.76846     
snp10003  Monomorphic -          -        -         -            -           
snp10004  Monomorphic -          -        -         -            -           
snp10005  -           0.63306    0.43881  0.50130   0.55427      0.37267     
snp10006  Monomorphic -          -        -         -            -           
snp10007  Monomorphic -          -        -         -            -           
snp10008  -           0.20627    0.30005  0.08652   0.83655      0.13493     
snp10009  -           0.74736    0.87204  0.47815   0.68152      0.93616     
snp100010 Monomorphic -          -        -         -            -           
snp100011 -           0.30538    0.12921  0.27289   0.28415      0.12787     
snp100012 -           0.70573    0.58360  0.47929   0.72341      0.48994     
snp100013 -           0.14971    0.09878  0.11040   0.38420      0.05475     
snp100014 -           0.03796    0.01511  0.11952   0.27139      0.01245     
snp100015 -           0.02626    -        -         -            -           
snp100016 Monomorphic -          -        -         -            -           
snp100017 -           0.70644    0.79126  0.45966   0.59972      0.99917     
          -        -         -            -      

----------------------------------

> myres2 <- WGassociation(protein,data=mySNP,model="all")
> myres2
          comments    codominant dominant recessive overdominant log-additive
snp10001  -           0.00586    0.00516  0.01605   0.12306      0.00140     
snp10002  -           0.78555    0.93303  0.48704   0.87288      0.76846     
snp10003  Monomorphic -          -        -         -            -           
snp10004  Monomorphic -          -        -         -            -           
snp10005  -           0.63306    0.43881  0.50130   0.55427      0.37267  



---------------


> dominant(myres)
 [1] 0.005164929 0.933029300          NA          NA 0.438808500          NA          NA 0.300054230 0.872038400
[10]          NA 0.129205100 0.583599200 0.098778790 0.015114890          NA          NA 0.791264500 0.886923300
[19] 0.007494522 0.313808950          NA          NA 0.995438100 0.017737798          NA          NA 0.714963100
[28] 0.005040410 0.021833942          NA          NA 0.006561046 0.004546931 0.021833942          NA
> 
> recessive(myres)
 [1] 0.016047565 0.487039400          NA          NA 0.501301800          NA          NA 0.086524570 0.478153600
[10]          NA 0.272885900 0.479292900 0.110399960 0.119523580          NA          NA 0.459659700 0.478153600
[19] 0.147754655 0.086524570          NA          NA 0.481948100 0.005475794          NA          NA 0.949004900
[28] 0.087582190 0.005475794          NA          NA 0.096555480 0.068387908 0.005475794          NA
> 
> WGstats(myres)
$snp10001

SNP: snp10001  adjusted by: 
               n    me   se    dif  lower   upper  p-value  AIC
Codominant                                                     
T/T           92 47419 2393      0                0.005861 3602
C/T           53 38987 3177  -8432 -16165  -698.2              
C/C           12 27413 6061 -20006 -33770 -6241.5              
Dominant                                                       
T/T           92 47419 2393      0                0.005165 3603
C/T-C/C       65 36851 2858 -10568 -17870 -3266.8              
Recessive                                                      
T/T-C/T      145 44337 1935      0                0.016048 3605
C/C           12 27413 6061 -16924 -30549 -3298.9              
Overdominant                                                   
T/T-C/C      104 45111 2308      0                0.123062 3608
C/T           53 38987 3177  -6123 -13864  1617.1              
log-Additive                                                   
0,1,2                        -9332 -14955 -3708.6 0.001404 3601

$snp10002

SNP: snp10002  adjusted by: 
               n    me   se    dif  lower upper p-value  AIC
Codominant                                                  
C/C           74 42876 2890    0.0               0.7855 3612
A/C           78 42740 2576 -135.8  -7648  7377             
A/A            5 50262 6879 7385.6 -14006 28777             
Dominant                                                    
C/C           74 42876 2890    0.0               0.9330 3611
A/C-A/A       83 43193 2456  317.3  -7072  7706             
Recessive                                                   
C/C-A/C      152 42806 1924    0.0               0.4870 3610
A/A            5 50262 6879 7455.3 -13518 28429             
Overdominant                                                
C/C-A/A       79 43343 2742    0.0               0.8729 3611
A/C           78 42740 2576 -603.2  -7980  6773             
log-Additive                                                
0,1,2                        996.5  -5626  7619  0.7685 3611

$snp10003

SNP: snp10003  adjusted by: 
Monomorphic

$snp10004

SNP: snp10004  adjusted by: 
Monomorphic

> summary(myres)
 SNPs (n) Genot error (%) Monomorphic (%) Significant* (n) (%)
       35               0            34.3                0   0

 *Number of statistically significant associations at level 1e-06
> plot(myres)

> resHapMap <- WGassociation(protein,data=mySNP,model="log")
> 
> resHapMap
          comments    log-additive
snp10001  -           0.00140     
snp10002  -           0.76846     
snp10003  Monomorphic -           
snp10004  Monomorphic -           
snp10005  -           0.37267


--------[p.163]----------------------------


The Whole Genome SNP Association Analysis


> data(HapMap)
> 
> str(HapMap)
'data.frame':   120 obs. of  9307 variables:
 $ id        : Factor w/ 120 levels "NA06985","NA06993",..: 1 2 3 4 5 6 7 8 9 10 ...
 $ group     : Factor w/ 2 levels "CEU","YRI": 1 1 1 1 1 1 1 1 1 1 ...
 $ rs10399749: Factor w/ 1 level "CC": 1 1 1 1 1 1 1 1 1 1 ...
 $ rs11260616: Factor w/ 3 levels "AA","AT","TT": 1 2 1 2 1 1 1 1 2 2 ...
 $ rs4648633 : Factor w/ 3 levels "CC","CT","TT": 3 2 3 3 2 3 3 2 2 NA ...
 $ rs6659552 : Factor w/ 3 levels "CC","CG","GG": 3 2 2 3 3 2 2 2 2 3 ...
 $ rs7550396 : Factor w/ 2 levels "AG","GG": 2 2 2 2 2 2 2 2 2 

> str(HapMap.SNPs.pos)
'data.frame':   9305 obs. of  3 variables:
 $ snp       : chr  "rs10399749" "rs11260616" "rs4648633" "rs6659552" ...
 $ chromosome: chr  "chr1" "chr1" "chr1" "chr1" ...
 $ position  : int  45162 1794167 2352864 2902617 3170389 3382844 3766548 4038743 4246260 4387131 ...
> 


> myhapmap <- setupSNP(HapMap,colSNPs=3:9307,sort=TRUE,info=HapMap.SNPs.pos,sep="")


  rs4864461 rs729474 rs11736469 rs11941906 rs13116039 rs11727673
1       C/C      C/C        C/C        T/T        G/G        T/T
2       C/C      C/C        C/C        T/T        G/G        T/T
3       C/C      C/G        C/C        T/T        G/G        T/T
4       C/C      C/G        C/C        T/T        G/G        A/T
5       C/C      C/C        C/C        T/T        G/G        T/T
6       C/C      C/G        C/C        T/T        G/G        T/T
  rs6554138 rs1105825 rs4864516 rs4318706 rs11133353 rs4356961
1       A/A       T/T       T/T       A/A       <NA>       C/C
2       A/G       T/T       C/C       A/T        A/A       C/C
3       A/G       T/T       C/T       A/A        A/A       C/C
4       A/G       T/T      <NA>       A/A        A/A       C/C
5       A/G       T/T       T/T       A/T        A/A       C/C


> res4<-WGassociation(group,data=myhapmap,model="dominant")
     5分位かかる
> head(res4)
           comments    dominant
rs10399749 Monomorphic -       
rs11260616 -           0.09851 
rs4648633  -           0.00000 
rs6659552  -           0.00000 
rs7550396  -           1.00000 
rs12239794 Monomorphic -       
> 
-------------------------------------------


> install.packages("GenABEL")
also installing the dependency ‘GenABEL.data’

 URL 'https://cran.ism.ac.jp/bin/windows/contrib/3.2/GenABEL.data_1.0.0.zip' を試しています 
Content type 'application/zip' length 2419647 bytes (2.3 MB)
downloaded 2.3 MB

 URL 'https://cran.ism.ac.jp/bin/windows/contrib/3.2/GenABEL_1.8-0.zip' を試しています 
Content type 'application/zip' length 3664426 bytes (3.5 MB)
downloaded 3.5 MB

 パッケージ ‘GenABEL.data’ は無事に展開され、MD5 サムもチェックされました 
 パッケージ ‘GenABEL’ は無事に展開され、MD5 サムもチェックされました 

 ダウンロードされたパッケージは、以下にあります 
        C:\Users\英里\AppData\Local\Temp\RtmpIT6uZ9\downloaded_packages 
> library(GenABEL)
 要求されたパッケージ MASS をロード中です 
 要求されたパッケージ GenABEL.data をロード中です 

>convert.snp.ped(pedfile="myPed.ped",mapfile="myMap.map",out="myOut.out")
Reading map from file 'myMap.map' ...
... done.  Read positions of 214 markers from file 'myMap.map'
Reading genotypes from file 'myPed.ped' ...
...done.  Read information for 90 people from file 'myPed.ped'
Analysing marker information ...
Writing to file 'myOut.out' ...
... done.

 
> mydat<-load.gwaa.data(phenofile="myPheno.ph",genofile="myOut.out")
ids loaded...
marker names loaded...
chromosome data loaded...
map data loaded...
allele coding data loaded...
strand data loaded...
genotype data loaded...
snp.data object created...
assignment of gwaa.data object FORCED; X-errors were not checked!
> head(str(mydat))
Formal class 'gwaa.data' [package "GenABEL"] with 2 slots
  ..@ phdata:'data.frame':      90 obs. of  3 variables:
  .. ..$ id : chr [1:90] "NA06985" "NA06991" "NA06993" "NA06994" ...
  .. ..$ sex: int [1:90] 1 0 0 0 1 1 0 0 1 1 ...
  .. ..$ aff: int [1:90] 1 0 1 0 1 0 0 1 0 1 ...
  ..@ gtdata:Formal class 'snp.data' [package "GenABEL"] with 11 slots
  .. .. ..@ nbytes    : num 23
  .. .. ..@ nids      : int 90
  .. .. ..@ nsnps     : int 214
  .. .. ..@ idnames   : chr [1:90] "NA06985" "NA06991" "NA06993" "NA06994" ...
  .. .. ..@ snpnames  : chr [1:214] "rs679153" "rs9965482" "rs16943926" "rs16943929" ...
  .. .. ..@ chromosome: Factor w/ 1 level "18": 1 1 1 1 1 1 1 1 1 1 ...
  .. .. .. ..- attr(*, "names")= chr [1:214] "rs679153" "rs9965482" "rs16943926" "rs16943929" ...
  .. .. ..@ map       : Named num [1:214] 2859916 2860891 2861667 2861811 2862091 ...
  .. .. .. ..- attr(*, "names")= chr [1:214] "rs679153" "rs9965482" "rs16943926" "rs16943929" ...
  .. .. ..@ coding    :Formal class 'snp.coding' [package "GenABEL"] with 1 slot
  .. .. .. .. ..@ .Data: raw [1:214] 17 1b 1b 1b ...

> phdata(mydat)
             id sex aff
NA06985 NA06985   1   1
NA06991 NA06991   0   0
NA06993 NA06993   0   1
NA06994 NA06994   0   0
NA07000 NA07000   1   1
NA07019 NA07019   1   0
NA07022 NA07022   0   0
NA07029 NA07029   0   1
NA07034 NA07034   1   0
NA07048 NA07048   1   1

> gtdata(mydat)
@nids = 90 
@nsnps = 214 
@nbytes = 23 
@idnames = NA06985 NA06991 NA06993 NA06994 NA07000 NA07019 NA07022 NA07029 NA07034 NA07048 NA07055 NA07056 NA07345 NA07348 NA07357 NA10830 NA10831 NA10835 NA10838 NA10839 NA10846 NA10847 NA10851 NA10854 NA10855 NA10856 NA10857 NA10859 NA10860 NA10861 NA10863 NA11829 NA11830 NA11831 NA11832 NA11839 NA11840 NA11881 NA11882 NA11992 NA11993 NA11994 NA11995 NA12003 NA12004 NA12005 NA12006 NA12043 NA12044 NA12056 NA12057 NA12144 NA12145 NA12146 NA12154 NA12155 NA12156 NA12234 NA12236 NA12239 NA12248 NA12249 NA12264 NA12707 NA12716 NA12717 NA12740 NA12750 NA12751 NA12752 NA12753 NA12760 NA12761 NA12762 NA12763 NA12801 NA12802 NA12812 NA12813 NA12814 NA12815 NA12864 NA12865 NA12872 NA12873 NA12874 NA12875 NA12878 NA12891 NA12892 
@snpnames = rs679153 rs9965482 rs16943926 rs16943929 rs612071 rs9950998 rs613796 rs623561 rs17629201 rs11663534 rs9960065 rs9951458 rs9962679 rs4797088 rs568289 rs17553620 rs583523 rs680173 rs642887 rs9955018 rs7226712 rs12454179 rs6506037 rs518378 rs16943962 rs16943963 rs8097282 rs8085419 rs637647 rs16943974 rs592120 rs4798040 rs3810067 rs16943981 rs3826638 rs3810066 rs551024 rs604050 rs16943997 rs7241203 rs11080994 rs659337 rs6506038 rs4371232 rs626936 rs12150778 rs16944003 rs12969432 rs11080995 rs4798043 rs4798044 rs10163804 rs4797091 rs609452 rs607411 rs12957670 rs1790994 rs1059281 rs8098776 rs3745013 rs642926 rs1164 rs1985 rs607549 rs2282635 rs2282636 rs17555442 rs16944051 rs16944053 rs11660930 rs3897689 rs3819090 rs602233 rs4797092 rs635500 rs618760 rs3765622 rs7237795 rs8094703 rs647195 rs9961969 rs9962809 rs670261 rs10853287 rs627538 rs613039 rs11876611 rs17556090 rs643507 rs641287 rs9675656 rs10460009 rs1785282 rs1628891 rs7244259 rs7244566 rs681670 rs588168 rs643015 rs8095489 rs603080 rs8096026 rs751375 rs1784748 rs652587 rs11662054 rs11877821 rs621405 rs1539791 rs10163589 rs660716 rs2298786 rs8087073 rs12457925 rs625527 rs635836 rs638331 rs618481 rs4121532 rs12454079 rs11876928 rs3897586 rs616117 rs2276377 rs2276376 rs635703 rs3934427 rs11663067 rs9967441 rs681869 rs680085 rs682853 rs8094920 rs589318 rs637854 rs11663615 rs650543 rs661247 rs610810 rs661767 rs661338 rs12456042 rs3016722 rs606574 rs11664104 rs16944160 rs10502310 rs17637670 rs665715 rs666624 rs666676 rs681043 rs650787 rs11661976 rs3016718 rs587721 rs11661932 rs584185 rs680813 rs7228587 rs678156 rs1784752 rs676739 rs665939 rs664607 rs16944186 rs16944189 rs16944193 rs9965612 rs16944195 rs4570954 rs4531822 rs16944200 rs2298541 rs8089690 rs7233117 rs3810063 rs3810061 rs6506041 rs9303915 rs10438875 rs10438876 rs4798054 rs4798055 rs12458507 rs8092807 rs9954551 rs7232721 rs7236788 rs4798058 rs9957978 rs4414562 rs9953654 rs1004413 rs9956108 rs16944238 rs1122568 rs4528660 rs4603644 rs16944246 rs1815804 rs7242138 rs17638424 rs4445969 rs4633799 rs7239382 rs4594324 rs4239320 rs7241663 rs11665427 rs7228981 rs6506052 rs12604162 rs9955849 
@chromosome = 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 
@coding =  17 1b 1b 1b 01 17 17 17 17 01 01 01 01 01 17 01 17 17 01 1b 17 17 01 1b 01 17 01 01 17 1b 17 1b 01 1b 17 1b 01 01 17 1b 17 1b 01 1b 1b 01 01 17 17 01 17 1b 01 1b 17 1b 17 01 1b 17 01 01 01 01 17 17 17 01 17 17 01 01 01 01 1b 01 17 01 1b 01 17 1b 1b 01 17 17 1b 01 1b 17 17 01 01 17 1b 1b 01 17 01 1b 17 17 17 1b 17 01 1b 01 17 1b 17 17 1b 1b 01 01 01 1b 1b 01 17 17 1b 1b 17 01 01 17 01 01 1b 1b 01 17 1b 1b 17 1b 1b 01 1b 17 1b 01 01 17 01 01 1b 17 1b 1b 1b 17 1b 01 17 1b 1b 01 1b 01 1b 1b 01 1b 1b 17 17 1b 17 1b 1b 01 1b 01 1b 1b 01 17 01 17 01 01 1b 01 1b 1b 17 01 01 01 1b 01 01 01 17 17 01 17 17 17 01 01 01 17 1b 17 17 17 01 17 01 17 
@strand =  00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 
@map = 2859916 2860891 2861667 2861811 2862091 2862368 2862449 2862586 2862679 2862721 2862821 2863083 2863119 2863848 2863900 2863939 2864156 2864364 2864408 2864431 2864552 2865255 2865432 2865893 2866438 2866875 2868613 2868647 2871839 2874667 2875118 2880942 2881478 2881490 2881847 2882201 2883172 2883413 2884164 2885903 2887993 2888131 2888535 2888563 2888677 2889184 2889324 2889428 2890028 2890639 2890671 2890725 2891007 2898482 2899831 2902506 2903431 2903896 2905068 2905219 2906054 2907152 2907223 2909606 2911444 2911777 2912149 2912716 2912900 2912996 2916054 2916402 2917023 2920138 2920765 2922189 2924536 2925022 2925877 2926036 2928147 2928238 2928839 2931377 2931450 2932437 2933278 2933795 2936291 2936833 2937220 2938029 2938476 2938632 2940744 2940944 2941876 2942799 2942988 2943609 2943798 2947166 2949758 2950457 2951033 2952177 2952338 2953479 2953695 2953948 2956840 2957456 2957845 2958216 2958752 2959355 2959374 2962981 2963483 2965061 2966481 2968689 2968896 2969038 2969173 2969489 2969828 2970122 2970374 2971398 2972439 2973053 2973306 2973942 2974281 2975072 2977038 2977103 2978767 2980700 2980800 2981225 2981411 2983695 2984861 2985308 2985373 2986922 2987661 2987877 2987907 2988764 2988890 2988950 2989261 2990909 2991286 2991658 2991861 2992230 2992436 2992455 2992764 2992875 2992922 2994490 2994645 2994756 2994981 2995931 2996207 2997083 3000266 3000450 3000565 3000609 3002019 3002144 3005144 3005919 3006126 3006150 3007136 3007152 3008467 3013264 3017680 3021319 3022293 3026349 3027610 3029186 3031403 3032297 3032416 3033406 3033505 3033516 3034269 3034867 3035429 3035739 3038453 3038871 3039472 3040737 3041286 3041297 3043309 3043491 3045433 3047213 3048407 3052581 
@male = 1 0 0 0 1 1 0 0 1 1 0 1 1 0 0 0 1 1 0 0 1 1 1 1 0 0 0 1 0 1 1 0 0 0 1 0 0 0 1 1 1 0 1 1 1 1 0 1 1 0 0 0 0 0 0 0 1 1 0 1 1 0 0 0 1 1 0 1 0 0 0 0 0 0 0 1 1 0 1 1 1 0 1 0 0 0 0 1 0 1 
@gtps = 
80 40 40 40 80 40 80 80 40 40 40 40 40 40 80 40 40 80 40 40 c0 40 80 40 40 c0 c0 c0 40 40 80 40 40 40 80 40 40 80 40 40 80 40 80 40 40 40 40 80 40 40 80 40 40 40 40 40 40 80 40 80 40 40 80 40 80 80 80 40 40 00 80 80 c0 40 40 c0 80 80 40 c0 80 40 40 40 c0 c0 40 80 40 80 80 40 c0 80 40 40 c0 c0 c0 40 c0 c0 c0 40 c0 80 40 c0 40 00 c0 40 40 40 c0 c0 c0 40 40 40 40 40 40 40 c0 c0 80 80 40 c0 40 40 80 c0 40 40 c0 40 40 c0 40 40 40 c0 c0 40 40 80 40 c0 00 40 40 80 40 40 80 40 40 80 40 40 40 40 c0 40 40 40 40 40 80 40 40 40 40 40 40 40 40 80 80 80 40 40 40 80 40 40 80 80 80 80 40 80 40 40 80 40 80 80 80 80 40 80 40 80 40 80 80 80 40 80 40 40 
40 40 40 40 40 40 40 c0 40 40 40 40 40 40 c0 80 40 80 40 40 c0 40 80 40 40 c0 c0 c0 40 40 80 40 40 40 80 40 40 80 40 40 40 40 80 40 40 80 40 80 80 80 40 40 80 40 40 40 40 c0 40 c0 40 40 c0 40 40 40 40 40 40 40 40 40 80 40 40 80 40 40 40 80 40 40 40 40 80 80 40 40 40 40 80 40 80 40 40 40 80 80 80 40 80 80 80 40 80 40 40 80 00 40 80 40 40 40 80 80 80 40 40 40 40 40 40 40 80 80 80 80 40 80 40 40 40 80 40 40 80 40 40 80 40 40 40 80 80 40 40 40 40 80 40 40 40 00 40 40 40 40 40 80 40 40 40 40 80 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 c0 40 40 80 80 80 80 40 80 40 40 80 40 80 80 80 80 40 80 40 80 40 80 80 80 40 80 40 40 
40 40 40 40 40 40 40 c0 40 40 40 40 40 40 80 80 40 80 40 40 80 80 80 40 80 80 80 80 40 40 80 00 40 40 40 40 40 40 40 40 80 40 40 40 40 80 40 80 80 80 80 40 80 40 40 40 40 c0 40 c0 40 40 00 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 00 40 40 40 40 40 40 40 40 40 40 00 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 40 80 80 80 40 40 40 80 40 40 80 40 40 40 40 40 40 40 40 40 40 40 40 40 40 80 40 00 40 80 80 80 00 80 80 40 
c0 40 40 40 c0 40 c0 40 40 40 40 40

> write.csv(phdata(mydat),file="tmppheno.ph")


```
