```
packageの依存性を管理するパッケージ

> install.packages("forecast", repos=c("http://rstudio.org/_packages", "http://cran.rstudio.com")) 
Installing package into ‘C:/Users/ekaminuma/Documents/R/win-library/3.2’
(as ‘lib’ is unspecified)
also installing the dependencies ‘timeDate’, ‘tseries’, ‘fracdiff’, ‘RcppArmadillo’



---[p.255]----------------------------------------------------
Enriching RNA-seq data with GO terms

https://bioconductor.org/packages/devel/bioc/manuals/goseq/man/goseq.pdf


> source("https://bioconductor.org/biocLite.R")
Bioconductor version 3.1 (BiocInstaller 1.18.5), ?biocLite for help
A newer version of Bioconductor is available for this version of R, ?BiocUpgrade for help
> 
> biocLite("goseq")
BioC_mirror: http://bioconductor.org
Using Bioconductor version 3.1 (BiocInstaller 1.18.5), R version 3.2.5.
Installing package(s) ‘goseq’
also installing the dependency ‘geneLenDataBase



> library(goseq)
> library(edgeR)
> supportedGenomes()
          db             species       date
1       hg38               Human  Dec. 2013
2       hg19               Human  Feb. 2009
3       hg18               Human  Mar. 2006
4       hg17               Human   May 2004
5       hg16               Human  Jul. 2003
6    vicPac2              Alpaca  Mar. 2013
7    vicPac1              Alpaca  Jul. 2008
8    dasNov3           Armadillo  Dec. 2011
9    papHam1              Baboon  Nov. 2008
10   panPan1              Bonobo   May 2012
11   aptMan1          Brown kiwi  Jun. 2015
12   otoGar3            Bushbaby  Mar. 2011
13   papAnu2            Bushbaby  Mar. 2012
14   felCat8                 Cat  Nov. 2014
15   felCat5                 Cat  Sep. 2011
16   felCat4                 Cat  Dec. 2008
17   felCat3                 Cat  Mar. 2006
18   panTro4               Chimp  Feb. 2011
19   panTro3               Chimp  Oct. 2010
20   panTro2               Chimp  Mar. 2006
21   panTro1               Chimp  Nov. 2003
22   criGri1     Chinese hamster  Jul. 2013
23   bosTau8                 Cow  Jun. 2014
24   bosTau7                 Cow  Oct. 2011
25   bosTau6                 Cow  Nov. 2009
26   bosTau4                 Cow  Oct. 2007
27   bosTau3                 Cow  Aug. 2006
28   bosTau2                 Cow  Mar. 2005
29   macFas5 Crab-eating macaque  Jun. 2013
30   canFam3                 Dog  Sep. 2011
31   canFam2                 Dog   May 2005
32   canFam1                 Dog  Jul. 2004
33   turTru2             Dolphin  Oct. 2011
34   loxAfr3            Elephant  Jul. 2009
35   musFur1              Ferret  Apr. 2011
36   nomLeu3              Gibbon  Oct. 2012
37   nomLeu2              Gibbon  Jun. 2011
38   nomLeu1              Gibbon  Jan. 2010
39   gorGor4             Gorilla   Dec 2014
40   gorGor3             Gorilla   May 2011
41   cavPor3          Guinea pig  Feb. 2008
42   eriEur2            Hedgehog   May 2012
43   eriEur1            Hedgehog  Jun. 2006
44   equCab2               Horse  Sep. 2007
45   equCab1               Horse  Jan. 2007
46   dipOrd1        Kangaroo rat  Jul. 2008
47   triMan1             Manatee  Oct. 2011
48   calJac3            Marmoset  Mar. 2009
49   calJac1            Marmoset  Jun. 2007
50   pteVam1             Megabat  Jul. 2008
51   myoLuc2            Microbat  Jul. 2010
52   balAcu1         Minke whale  Oct. 2013
53      mm10               Mouse  Dec. 2011
54       mm9               Mouse  Jul. 2007
55       mm8               Mouse  Feb. 2006
56       mm7               Mouse  Aug. 2005
57   micMur2         Mouse lemur   May 2015
58   micMur1         Mouse lemur  Jul. 2007
59   hetGla2      Naked mole-rat  Jan. 2012
60   hetGla1      Naked mole-rat  Jul. 2011
61   monDom5             Opossum  Oct. 2006
62   monDom4             Opossum  Jan. 2006
63   monDom1             Opossum  Oct. 2004
64   ponAbe2           Orangutan  Jul. 2007
65   ailMel1               Panda  Dec. 2009
66   susScr3                 Pig  Aug. 2011
67   susScr2                 Pig  Nov. 2009
68   ochPri3                Pika   May 2012
69   ochPri2                Pika  Jul. 2008
70   ornAna2            Platypus  Feb. 2007
71   ornAna1            Platypus  Mar. 2007
72   oryCun2              Rabbit  Apr. 2009
73       rn6                 Rat  Jul. 2014
74       rn5                 Rat  Mar. 2012
75       rn4                 Rat  Nov. 2004
76       rn3                 Rat  Jun. 2003
77   rheMac8              Rhesus  Nov. 2015
78   rheMac3              Rhesus  Oct. 2010
79   rheMac2              Rhesus  Jan. 2006
80   proCap1          Rock hyrax  Jul. 2008
81   oviAri3               Sheep  Aug. 2012
82   oviAri1               Sheep  Feb. 2010
83   sorAra2               Shrew  Aug. 2008
84   sorAra1               Shrew  Jun. 2006
85   choHof1               Sloth  Jul. 2008
86   speTri2            Squirrel  Nov. 2011
87   saiBol1     Squirrel monkey  Oct. 2011
88   tarSyr2             Tarsier  Sep. 2013
89   tarSyr1             Tarsier  Aug. 2008
90   sarHar1     Tasmanian devil  Feb. 2011
91   echTel2              Tenrec  Nov. 2012
92   echTel1              Tenrec  Jul. 2005
93   tupBel1          Tree shrew  Dec. 2006
94   macEug2             Wallaby  Sep. 2009
95   cerSim1    White rhinoceros   May 2012
96   allMis1  American alligator  Aug. 2012
97   gadMor1        Atlantic cod   May 2010
98   melUnd1          Budgerigar  Sep. 2011
99   galGal4             Chicken Nov.\n2011
100  galGal3             Chicken   May 2006
101  galGal2             Chicken  Feb. 2004
102  latCha1          Coelacanth  Aug. 2011
103  calMil1      Elephant shark  Dec. 2013
104      fr3                Fugu  Oct. 2011
105      fr2                Fugu  Oct. 2004
106      fr1                Fugu  Aug. 2002
107  petMar2             Lamprey  Sep. 2010
108  petMar1             Lamprey  Mar. 2007
109  anoCar2              Lizard   May 2010
110  anoCar1              Lizard  Feb. 2007
111  oryLat2              Medaka  Oct. 2005
112  geoFor1 Medium ground finch  Apr. 2012
113  oreNil2        Nile tilapia  Jan. 2011
114  chrPic1      Painted turtle  Dec. 2011
115  gasAcu1         Stickleback  Feb. 2006
116  tetNig2           Tetraodon  Mar. 2007
117  tetNig1           Tetraodon  Feb. 2004
118  melGal1              Turkey  Dec. 2009
119  xenTro7       X. tropicalis  Sep. 2012
120  xenTro3       X. tropicalis  Nov. 2009
121  xenTro2       X. tropicalis  Aug. 2005
122  xenTro1       X. tropicalis  Oct. 2004
123  taeGut2         Zebra finch  Feb. 2013
124  taeGut1         Zebra finch  Jul. 2008
125 danRer10           Zebrafish  Sep. 2014
126  danRer7           Zebrafish  Jul. 2010
127  danRer6           Zebrafish  Dec. 2008
128  danRer5           Zebrafish  Jul. 2007
129  danRer4           Zebrafish  Mar. 2006
130  danRer3           Zebrafish   May 2005
131      ci2     C. intestinalis  Mar. 2005
132      ci1     C. intestinalis  Dec. 2002
133  braFlo1            Lancelet  Mar. 2006
134  strPur2       S. purpuratus  Sep. 2006
135  strPur1       S. purpuratus  Apr. 2005
136  apiMel2        A. mellifera  Jan. 2005
137  apiMel1        A. mellifera  Jul. 2004
138  anoGam1          A. gambiae  Feb. 2003
139  droAna2        D. ananassae  Aug. 2005
140  droAna1        D. ananassae  Jul. 2004
141  droEre1           D. erecta  Aug. 2005
142  droGri1        D. grimshawi  Aug. 2005
143      dm6     D. melanogaster  Aug. 2014
144      dm3     D. melanogaster  Apr. 2006
145      dm2     D. melanogaster  Apr. 2004
146      dm1     D. melanogaster  Jan. 2003
147  droMoj2       D. mojavensis  Aug. 2005
148  droMoj1       D. mojavensis  Aug. 2004
149  droPer1       D. persimilis  Oct. 2005
150      dp3    D. pseudoobscura  Nov. 2004
151      dp2    D. pseudoobscura  Aug. 2003
152  droSec1        D. sechellia  Oct. 2005
153  droSim1         D. simulans  Apr. 2005
154  droVir2          D. virilis  Aug. 2005
155  droVir1          D. virilis  Jul. 2004
156  droYak2           D. yakuba  Nov. 2005
157  droYak1           D. yakuba  Apr. 2004
158   caePb2         C. brenneri  Feb. 2008
159   caePb1         C. brenneri  Jan. 2007
160      cb3         C. briggsae  Jan. 2007
161      cb1         C. briggsae  Jul. 2002
162     ce11          C. elegans  Feb. 2013
163     ce10          C. elegans  Oct. 2010
164      ce6          C. elegans   May 2008
165      ce4          C. elegans  Jan. 2007
166      ce2          C. elegans  Mar. 2004
167  caeJap1         C. japonica  Mar. 2008
168  caeRem3          C. remanei   May 2007
169  caeRem2          C. remanei  Mar. 2006
170  priPac1        P. pacificus  Feb. 2007
171  aplCal1            Sea Hare  Sep. 2008
172  sacCer3               Yeast April 2011
173  sacCer2               Yeast  June 2008
174  sacCer1               Yeast  Oct. 2003
175  eboVir3         Ebola Virus  June 2014
                                                          name
1                           Genome Reference Consortium GRCh38
2                           Genome Reference Consortium GRCh37
3                                              NCBI Build 36.1
4                                                NCBI Build 35
5                                                NCBI Build 34
6                          Broad Institute Vicugna_pacos-2.0.1
7                                    Broad Institute VicPac1.0
8                                      Broad Institute DasNov3
9                     Baylor College of Medicine HGSC Pham_1.0
10                                Max-Planck Institute panpan1
11  Max-Plank Institute for Evolutionary Anthropology AptMant0
12                                     Broad Institute OtoGar3
13                         Baylor College of Medicine Panu_2.0
14                                       ICGSC Felis_catus_8.0
15                                       ICGSC Felis_catus-6.2
16                                            NHGRI catChrV17e
17                                   Broad Institute Release 3
18                                            CGSC Build 2.1.4
19                                            CGSC Build 2.1.3
20                                              CGSC Build 2.1
21                                              CGSC Build 1.1
22        Beijing Genomics Institution-Shenzhen C_griseus_v1.0
23                               University of Maryland v3.1.1
24                  Baylor College of Medicine HGSC Btau_4.6.1
25                                 University of Maryland v3.1
26                    Baylor College of Medicine HGSC Btau_4.0
27                    Baylor College of Medicine HGSC Btau_3.1
28                    Baylor College of Medicine HGSC Btau_2.0
29               Washington University Macaca_fascicularis_5.0
30                                        Broad Institute v3.1
31                                        Broad Institute v2.0
32                                        Broad Institute v1.0
33                         Baylor College of Medicine Ttru_1.4
34                                     Broad Institute LoxAfr3
35            Ferret Genome Sequencing Consortium MusPutFur1.0
36                 Gibbon Genome Sequencing Consortium Nleu3.0
37                 Gibbon Genome Sequencing Consortium Nleu1.1
38                 Gibbon Genome Sequencing Consortium Nleu1.0
39                     Wellcome Trust Sanger Institute gorGor4
40                   Wellcome Trust Sanger Institute gorGor3.1
41                                     Broad Institute cavPor3
42                                   Broad Institute EriEur2.0
43                                    Broad Institute Draft_v1
44                                     Broad Institute EquCab2
45                                     Broad Institute EquCab1
46                            Baylor/Broad Institute DipOrd1.0
47                                Broad Institute TriManLat1.0
48                               WUSTL Callithrix_jacchus-v3.2
49                             WUSTL Callithrix_jacchus-v2.0.2
50                                   Broad Institute Ptevap1.0
51                                   Broad Institute MyoLuc2.0
52                                             KORDI BalAcu1.0
53                          Genome Reference Consortium GRCm38
54                                               NCBI Build 37
55                                               NCBI Build 36
56                                               NCBI Build 35
57                             Baylor/Broad Institute Mmur_2.0
58                                   Broad Institute MicMur1.0
59                           Broad Institute HetGla_female_1.0
60                       Beijing Genomics Institute HetGla_1.0
61                             Broad Institute release MonDom5
62                             Broad Institute release MonDom4
63                             Broad Institute release MonDom1
64                                   WUSTL Pongo_albelii-2.0.2
65                                     BGI-Shenzhen AilMel 1.0
66              Swine Genome Sequencing Consortium Sscrofa10.2
67               Swine Genome Sequencing Consortium Sscrofa9.2
68                                   Broad Institute OchPri3.0
69                                     Broad Institute OchPri2
70                                                WUSTL v5.0.1
71                                                WUSTL v5.0.1
72                             Broad Institute release OryCun2
73                                               RGSC Rnor_6.0
74                                               RGSC Rnor_5.0
75                        Baylor College of Medicine HGSC v3.4
76                        Baylor College of Medicine HGSC v3.1
77                  Baylor College of Medicine HGSC Mmul_8.0.1
78                           Beijing Genomics Institute CR_1.0
79            Baylor College of Medicine HGSC v1.0 Mmul_051212
80                   Baylor College of Medicine HGSC Procap1.0
81                                               ISGC Oar_v3.1
82                                         ISGC Ovis aries 1.0
83                                   Broad Institute SorAra2.0
84                                   Broad Institute SorAra1.0
85                                   Broad Institute ChoHof1.0
86                                   Broad Institute SpeTri2.0
87                                   Broad Institute SaiBol1.0
88                                WashU Tarsius_syrichta-2.0.1
89                             WUSTL/Broad Institute Tarsyr1.0
90               Wellcome Trust Sanger Institute Devil_refv7.0
91                                   Broad Institute EchTel2.0
92                                     Broad Institute echTel1
93                                   Broad Institute Tupbel1.0
94        Tammar Wallaby Genome Sequencing Consortium Meug_1.1
95                                Broad Institute CerSimSim1.0
96            Int. Crocodilian Genomes Working Group allMis0.2
97                                     Genofisk GadMor_May2010
98                                                  WUSTL v6.3
99                                      ICGC Gallus-gallus-4.0
100                                    WUSTL Gallus-gallus-2.1
101                                    WUSTL Gallus-gallus-1.0
102                                    Broad Institute LatCha1
103                             IMCB Callorhinchus_milli_6.1.3
104                                                   JGI v5.0
105                                                   JGI v4.0
106                                                   JGI v3.0
107                                                  WUGSC 7.0
108                                                 WUSTL v3.0
109                                    Broad Institute AnoCar2
110                                    Broad Institute AnoCar1
111                                                   NIG v1.0
112                                BGI GeoFor_1.0 / NCBI 13302
113                          Broad Institute Release OreNil1.1
114                        IPTGSC Chrysemys_picta_bellii-3.0.1
115                                Broad Institute Release 1.0
116                                               Genoscope v7
117                                               Genoscope v7
118                             Turkey Genome Consortium v2.01
119                                                  JGI v.7.0
120                                                  JGI v.4.2
121                                                  JGI v.4.1
122                                                  JGI v.3.0
123                                            WashU taeGut324
124                                               WUSTL v3.2.4
125                        Genome Reference Consortium GRCz10 
126                                      Sanger Institute Zv9 
127                                      Sanger Institute Zv8 
128                                      Sanger Institute Zv7 
129                                      Sanger Institute Zv6 
130                                      Sanger Institute Zv5 
131                                                   JGI v2.0
132                                                   JGI v1.0
133                                                   JGI v1.0
134                Baylor College of Medicine HGSC v. Spur 2.1
135                Baylor College of Medicine HGSC v. Spur_0.5
136                Baylor College of Medicine HGSC v.Amel_2.0 
137                Baylor College of Medicine HGSC v.Amel_1.2 
138                                                IAGP v.MOZ2
139                                  Agencourt Arachne release
140                                        TIGR Celera release
141                                  Agencourt Arachne release
142                                  Agencourt Arachne release
143                                   BDGP Release 6 + ISO1 MT
144                                             BDGP Release 5
145                                             BDGP Release 4
146                                             BDGP Release 3
147                                  Agencourt Arachne release
148                                  Agencourt Arachne release
149                                    Broad Institute release
150                                        Flybase Release 1.0
151                   Baylor College of Medicine HGSC Freeze 1
152                                Broad Institute Release 1.0
153                                          WUSTL Release 1.0
154                                  Agencourt Arachne release
155                                  Agencourt Arachne release
156                                          WUSTL Release 2.0
157                                          WUSTL Release 1.0
158                                                WUSTL 6.0.1
159                                                  WUSTL 4.0
160                                                  WUSTL Cb3
161                                      WormBase v. cb25.agp8
162                  C. elegans Sequencing Consortium WBcel235
163                                          WormBase v. WS220
164                                          WormBase v. WS190
165                                          WormBase v. WS170
166                                          WormBase v. WS120
167                                                WUSTL 3.0.2
168                                               WUSTL 15.0.1
169                                                  WUSTL 1.0
170                                                  WUSTL 5.0
171                                    Broad Release Aplcal2.0
172                                    SGD April 2011 sequence
173                                     SGD June 2008 sequence
174                                    SGD 1 Oct 2003 sequence
175                       Sierra Leone 2014 (G3683/KM034562.1)
:
<続く>
                                                                             

> dat1<-read.table("Li_sum.txt",header=TRUE,stringsAsFactors=FALSE,row.names=1)
> 
> head(dat1)
                lane1 lane2 lane3 lane4 lane5 lane6 lane8
ENSG00000215688     0     0     0     0     0     0     0
ENSG00000215689     0     0     0     0     0     0     0
ENSG00000220823     0     0     0     0     0     0     0
ENSG00000242499     0     0     0     0     0     0     0
ENSG00000224938     0     0     0     0     0     0     0
ENSG00000239242     0     0     0     0     0     0     0

> mytr <- factor(rep(c("Control","Treatment"),times=c(4,3)) )
> mytr
[1] Control   Control   Control   Control   Treatment Treatment Treatment
Levels: Control Treatment

> myDG<-DGEList(dat1,lib.size=colSums(dat1),group=mytr)

> myDG
An object of class "DGEList"
$counts
                lane1 lane2 lane3 lane4 lane5 lane6 lane8
ENSG00000215688     0     0     0     0     0     0     0
ENSG00000215689     0     0     0     0     0     0     0
ENSG00000220823     0     0     0     0     0     0     0
ENSG00000242499     0     0     0     0     0     0     0
ENSG00000224938     0     0     0     0     0     0     0
49501 more rows ...

$samples
          group lib.size norm.factors
lane1   Control  1178832            1
lane2   Control  1384945            1
lane3   Control  1716355            1
lane4   Control  1767927            1
lane5 Treatment  2127868            1
lane6 Treatment  2142158            1
lane8 Treatment   816171            1

> disp1<-estimateCommonDisp(myDG)
> disp1
An object of class "DGEList"
$counts
                lane1 lane2 lane3 lane4 lane5 lane6 lane8
ENSG00000215688     0     0     0     0     0     0     0
ENSG00000215689     0     0     0     0     0     0     0
ENSG00000220823     0     0     0     0     0     0     0
ENSG00000242499     0     0     0     0     0     0     0
ENSG00000224938     0     0     0     0     0     0     0
49501 more rows ...

$samples
          group lib.size norm.factors
lane1   Control  1178832            1
lane2   Control  1384945            1
lane3   Control  1716355            1
lane4   Control  1767927            1
lane5 Treatment  2127868            1
lane6 Treatment  2142158            1
lane8 Treatment   816171            1

$common.dispersion
[1] 0.05688364

$pseudo.counts
:
> mytest<- exactTest(disp1)
> mytest
An object of class "DGEExact"
$table
                logFC   logCPM PValue
ENSG00000215688     0 0.330418      1
ENSG00000215689     0 0.330418      1
ENSG00000220823     0 0.330418      1
ENSG00000242499     0 0.330418      1
ENSG00000224938     0 0.330418      1
49501 more rows ...

$comparison
[1] "Control"   "Treatment"

$genes
NULL

> mytag1<-as.integer(p.adjust(mytest$table$Pvalue[mytest$table$logFC!=0],method="BH")<0.05)
> 

                  



> source("https://bioconductor.org/biocLite.R")
Bioconductor version 3.1 (BiocInstaller 1.18.5), ?biocLite for help
A newer version of Bioconductor is available for this version of R, ?BiocUpgrade
  for help
> biocLite("geneLenDataBase")
BioC_mirror: http://bioconductor.org
Using Bioconductor version 3.1 (BiocInstaller 1.18.5), R version 3.2.5.
Installing package(s) ‘geneLenDataBase’

---[p.258]----------------
田中さん担当分

The KEGG enrichment of sequence data







---[p.260]------------------------------------------------------
Analyzing methylation data


> biocLite("methyAnalysis")
BioC_mirror: http://bioconductor.org
Using Bioconductor version 3.1 (BiocInstaller 1.18.5), R version 3.2.5.
Installing package(s) ‘methyAnalysis’

> biocLite("TxDb.Hsapiens.UCSC.hg19.knownGene")

> library(methyAnalysis)
 要求されたパッケージ grid をロード中です 
 要求されたパッケージ BiocGenerics をロード中です

> data(exampleMethyGenoSet)
> summary(exampleMethyGenoSet)
      Length        Class         Mode 
           1 MethyGenoSet           S4 
> slotNames(exampleMethyGenoSet)
[1] "history"           "locData"           "assayData"        
[4] "phenoData"         "featureData"       "experimentData"   
[7] "annotation"        "protocolData"      ".__classVersion__"
> head(locData(exampleMethyGenoSet)
+ )
GRanges object with 6 ranges and 0 metadata columns:
             seqnames               ranges strand
                <Rle>            <IRanges>  <Rle>
  cg17035109    chr21 [10882029, 10882029]      *
  cg06187584    chr21 [10883548, 10883548]      *
  cg12459059    chr21 [10884748, 10884748]      *
  cg25450479    chr21 [10884967, 10884967]      *
  cg23347501    chr21 [10884969, 10884969]      *
  cg03661019    chr21 [10885409, 10885409]      *
  -------
  seqinfo: 1 sequence from hg19 genome; no seqlengths
> dim(exprs(exampleMethyGenoSet))
[1] 4243    8
> 
> pData(exampleMethyGenoSet)
        SampleType
Sample1      Type1
Sample2      Type1
Sample3      Type1
Sample4      Type1
Sample5      Type2
Sample6      Type2
Sample7      Type2
Sample8      Type2
 警告メッセージ: 
'pData' is deprecated.
Use 'colData' instead.
See help("Deprecated") 

> str(exampleMethyGenoSet)
Formal class 'MethyGenoSet' [package "methyAnalysis"] with 9 slots
  ..@ history          :'data.frame':	8 obs. of  4 variables:
  .. ..$ submitted  : chr [1:8] "2011-03-30 20:52:40" "2011-03-30 21:19:32" "2011-04-11 15:27:15" "2011-04-11 15:30:45" ...
  .. ..$ finished   : chr [1:8] "2011-03-30 21:19:32" "2011-03-30 21:21:22" "2011-04-11 15:29:27" "2011-04-11 15:32:36" ...

> sm <- smoothMethyData(exampleMethyGenoSet,winSize=200)
Smoothing Chromosome chr21 ...
> sm
MethyGenoSet (storageMode: lockedEnvironment)
assayData: 4243 features, 8 samples 
  element names: detection, exprs, methylated, unmethylated 
protocolData: none
phenoData
  sampleNames: Sample1 Sample2 ... Sample8 (8 total)
  varLabels: SampleType
  varMetadata: labelDescription
featureData
  featureNames: cg17035109 cg06187584 ... cg08821909 (4243
    total)
  fvarLabels: ID
  fvarMetadata: labelDescription
experimentData: use 'experimentData(object)'
Annotation: IlluminaHumanMethylation450k.db 
Feature Locations:
GRanges object with 4243 ranges and 0 metadata columns:
             seqnames               ranges strand
                <Rle>            <IRanges>  <Rle>
  cg17035109    chr21 [10882029, 10882029]      *
  cg06187584    chr21 [10883548, 10883548]      *
  cg12459059    chr21 [10884748, 10884748]      *
  cg25450479    chr21 [10884967, 10884967]      *
  cg23347501    chr21 [10884969, 10884969]      *
         ...      ...                  ...    ...
  cg14365143    chr21 [48088046, 48088046]      *
  cg22544673    chr21 [48088150, 48088150]      *
  cg02984790    chr21 [48090128, 48090128]      *
  cg07468397    chr21 [48090508, 48090508]      *
  cg08821909    chr21 [48091011, 48091011]      *
  -------
  seqinfo: 1 sequence from hg19 genome; no seqlengths

> attr(sm,'windowSize')
[1] 200

> cond <- pData(exampleMethyGenoSet)$SampleType
 警告メッセージ: 
'pData' is deprecated.
Use 'colData' instead.
See help("Deprecated") 
> 
> cond <- colData(exampleMethyGenoSet)$SampleType
> 
> myDMR <- detectDMR.slideWin(exampleMethyGenoSet,sampleType=cond)
Smoothing Chromosome chr21 ...
> 
> head(myDMR)
GRanges object with 6 ranges and 11 metadata columns:
             seqnames               ranges strand |    PROBEID
                <Rle>            <IRanges>  <Rle> |   <factor>
  cg17035109    chr21 [10882029, 10882029]      * | cg17035109
  cg06187584    chr21 [10883548, 10883548]      * | cg06187584
  cg12459059    chr21 [10884748, 10884748]      * | cg12459059
  cg25450479    chr21 [10884967, 10884967]      * | cg25450479
  cg23347501    chr21 [10884969, 10884969]      * | cg23347501
  cg03661019    chr21 [10885409, 10885409]      * | cg03661019
             difference    p.value  p.adjust     tscore
              <numeric>  <numeric> <numeric>  <numeric>
  cg17035109 -1.8411605 0.06276449 0.1888488 -2.2804091
  cg06187584 -0.4566059 0.41601486 0.6149596 -0.8734252
  cg12459059 -0.3591179 0.36542152 0.5627904 -0.9789270
  cg25450479 -0.3591179 0.36542152 0.5627904 -0.9789270
  cg23347501 -0.3591179 0.36542152 0.5627904 -0.9789270
  cg03661019 -0.3532662 0.38065600 0.5782099 -0.9460314
             startWinIndex endWinIndex startLocation endLocation
                 <numeric>   <numeric>     <integer>   <integer>
  cg17035109             1           1      10882029    10882029
  cg06187584             2           2      10883548    10883548
  cg12459059             3           5      10884748    10884969
  cg25450479             3           5      10884748    10884969
  cg23347501             3           5      10884748    10884969
  cg03661019             6           6      10885409    10885409
             mean_Type1  mean_Type2
              <numeric>   <numeric>
  cg17035109 -2.4183775 -0.57721699
  cg06187584 -2.2297567 -1.77315084
  cg12459059  0.2594151  0.61853304
  cg25450479  0.2594151  0.61853304
  cg23347501  0.2594151  0.61853304
  cg03661019 -0.4170363 -0.06377013
  -------
  seqinfo: 1 sequence from hg19 genome; no seqlengths

> SigDMR <- identifySigDMR(myDMR)
> dmr_ann <- annotateDMRInfo(SigDMR,'TxDb.Hsapiens.UCSC.hg19.knownGene')
Loading required package: TxDb.Hsapiens.UCSC.hg19.knownGene
Loading required package: GenomicFeatures
> 
> export.DMRInfo(dmr_anno,savePrefix='testExample')
 as.data.frame(DMRInfo.ann$sigDataInfo) でエラー: 
   引数 'x' の評価中にエラーが起きました (関数 'as.data.frame' に対するメソッドの選択時):  エラー:  オブジェクト 'dmr_anno' がありません 
 
> export.DMRInfo(dmr_ann,savePrefix='test_dmr_ann')

PCのterminalでファイルを確認する。

na-no-MacBook-Pro:~ ekaminum$ ls -lt

total 32

-rw-r--r--   1 ekaminum  staff  2450  7 13 10:47 DMRInfo_test_dmr_ann_2016-07-13.csv

-rw-r--r--   1 ekaminum  staff   420  7 13 10:47 DMRInfo_test_dmr_ann_2016.07.13.bed

-rw-r--r--   1 ekaminum  staff  4491  7 13 10:47 DMRdata_test_dmr_ann_2016-07-13.csv

na-no-MacBook-Pro:~ ekaminum$ cat DMRInfo_test_dmr_ann_2016-07-13.csv 

"CHROMOSOME","start","end","NumOfProbe","min_p.value","max_difference","min_p.adjust","max_tscore","mean_Type1","mean_Type2","Transcript","EntrezID","GeneSymbol","distance2TSS","nearestTx","PROMOTER"

"chr21",19191045,19191320,3,0.0001235924,-1.1907873545,0.007061388,-8.74702112,-4.02704733856667,-2.92264438366667,"uc002yko.4","54149","C21orf91",383,"uc002yko.4",FALSE

"chr21",34522588,34522588,1,0.000007677523,-1.2848047397,0.003093883,-14.182749104,-4.5115552816,-3.226750542,"uc002yra.4","728409","LINC01548",19953,"uc002yra.4",FALSE

"chr21",37851847,37851847,1,0.00009416121,-1.4922516167,0.006768435,-9.179626306,0.6201243857,2.112376002,"uc002yvl.1","23562","CLDN14",541,"uc002yvl.1",FALSE

na-no-MacBook-Pro:~ ekaminum$ cat DMRInfo_test_dmr_ann_2016.07.13.bed 

chr21	19191044	19191320	.	0	.

chr21	34522587	34522588	.	0	.

chr21	37851846	37851847	.	0	.

chr21	38066046	38066047	.	0	.

chr21	38075598	38077042	.	0	.

chr21	38080974	38081193	.	0	.

chr21	39285678	39285679	.	0	.

chr21	39748802	39748803	.	0	.

na-no-MacBook-Pro:~ ekaminum$ cat DMRdata_test_dmr_ann_2016-07-13.csv 

"CHROMOSOME","POSITION","PROBEID","difference","p.value","p.adjust","tscore","startWinIndex","endWinIndex","mean_Type1","mean_Type2","Transcript","EntrezID","GeneSymbol","distance2TSS","nearestTx","PROMOTER"

"chr21",19191096,"cg12430776",-1.07152346805949,0.000123592435561639,0.00706138813614358,-8.7470211196085,279,281,-4.09227128083324,-3.02074781277375,"uc002yko.4","54149","C21orf91",607,"uc002yko.4",FALSE

"chr21",34522588,"ch.21.33444458F",-1.28480473970052,7.6775226703329e-06,0.00309388312095657,-14.1827491036894,998,998,-4.51155528155554,-3.22675054185502,"uc002yra.4","728409","LINC01548",19953,"uc002yra.4",FALSE

```
